packages/backend/src/workers/provider/LlamaCppPython.ts

```diff
@@ -197,9 +197,21 @@ export class LlamaCppPython extends InferenceProvider {
     if (this.configurationRegistry.getExtensionConfiguration().experimentalGPU) {
       const gpus: IGPUInfo[] = await this.gpuManager.collectGPUs();
       if (gpus.length === 0) throw new Error('no gpu was found.');
-      if (gpus.length > 1)
-        console.warn(`found ${gpus.length} gpus: using multiple GPUs is not supported. Using ${gpus[0].model}.`);
-      gpu = gpus[0];
+      let selectedGPU = 0;
+      if (gpus.length > 1) {
+        // Look for a GPU that is of a known type, use the first one found.
+        // Fall back to the first one if no GPUs are of known type.
+        for (let i = 0; i < gpus.length; i++) {
+          if (gpus[i].vendor !== GPUVendor.UNKNOWN) {
+            selectedGPU = i;
+            break;
+          }
+        }
+        console.warn(
+          `found ${gpus.length} gpus: using multiple GPUs is not supported. Using ${gpus[selectedGPU].model}.`,
+        );
+      }
+      gpu = gpus[selectedGPU];
```

Comment:
```suggestion
      if (gpus.length > 1) {
        // Look for a GPU that is of a known type, use the first one found.
        // Fall back to the first one if no GPUs are of known type.
        gpu = gpus.find(({vendor}) => vendor !== GPUVendor.UNKNOWN) ?? gpus[0];
      }
```

Reply:
The change missed the case with 0, cpus. pulled it in minus the check for number of cpus and it seems to do the right thing. Also rebaseed.

Reply:
It is a lot more compact which is great.